### PR TITLE
[WIP] Refactor Post.__init__

### PR DESCRIPTION
**Pull request by ralsina**
Not a functional change, just break the monster into bits

### CODE REVIEW

File: nikola/post.py [hunk 1]
```diff
@@ -307,7 +305,6 @@ def __init__(
                     self.has_oldstyle_metadata_tags = True
 
         # While draft comes from the tags, it's not really a tag
```
Comment (Kwpolska) [inline nikola/post.py:307]: I think that’s a leftover comment from before the `status` meta key?

Comment (ralsina) [reply nikola/post.py:295]: Probably. Have not gotten that far yet

File: nikola/post.py [hunk 2]
```diff
@@ -92,176 +102,53 @@ def __init__(
...
-
-        if 'updated' not in default_metadata:
-            default_metadata['updated'] = default_metadata.get('date', None)
-
-        self.updated = to_datetime(default_metadata['updated'], tzinfo)
-
-        if 'title' not in default_metadata or 'slug' not in default_metadata \
-                or 'date' not in default_metadata:
+        # These are the required metadata fields
+        if 'title' not in default_metadata or 'slug' not in default_metadata:
```
Comment (Kwpolska) [inline nikola/post.py:137]: if `date` was removed from the required metadata set, then maybe we could remove it from the message as well?

**Comment by ralsina** [comment]
If tests pass, this is about as much as I can stomach refactoring it right now.

Further refining of each smaller chunk will be done in the future.

**Comment by ralsina** [comment]
Yes, good point.

File: nikola/post.py (hunk 2 above)
Comment (ralsina) [reply nikola/post.py:139]: Hmmm I am not sure I did the right thing here. Testing it.

Comment (ralsina) [reply nikola/post.py:139]: Confirmed, this is ok, it fails like this without ever reaching that message:

```
[2020-04-13T15:40:35Z] ERROR: Nikola: ValueError: Missing date in file posts/foo.rst
```

**Pull request merged**